app/(signed-in)/bookings/page.tsx

```diff
@@ -1,7 +1,97 @@
-import React from "react";
+"use client";
+
+import React, { useEffect, useState } from "react";
+import { useSession } from "next-auth/react";
+import {
+  PartialReservation,
+  retrieveAllReservations,
+} from "@/server-actions/reservation-actions";
+import Link from "next/link";
 
 const BookingsPage = () => {
-  return <div>BookingsPage</div>;
+  const { data: session, status } = useSession();
+  const [reservations, setReservations] = useState<PartialReservation[] | []>(
+    []
+  );
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchReservations = async () => {
+      if (!session || !session.user.email) {
+        setLoading(false);
+        setError("You must be logged in to view your reservations.");
+        return;
+      }
+
+      try {
+        const fetchedReservations = await retrieveAllReservations(
+          session.user.email
+        );
+        setReservations(fetchedReservations);
+      } catch (error: any) {
+        console.error("Error retrieving reservations:", error.message);
+        setError("Failed to load reservations.");
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    if (status === "authenticated") {
+      fetchReservations();
+    }
+  }, [session, status]);
+
+  if (loading) {
+    return <div>Loading your reservations...</div>;
+  }
+
+  if (error) {
+    return <div>{error}</div>;
+  }
+
+  if (reservations.length === 0) {
+    return <div>You have no reservations.</div>;
+  }
+
+  return (
+    <div>
+      <h1>Your Reservations</h1>
+      <ul className="list-disc pl-6">
+        {reservations.map((reservation) => (
+          <li key={reservation.bookingId} className="mb-4">
```

Comment:
Great stopping point for this PR.
Are you planning to turn this into cards in the next PR?

Reply:
Hmm, maybe. I would need to retrieve hotel room info through the API again to get pretty pictures.